presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -523,20 +527,34 @@ private PlanRoot createLogicalPlan()
             stateMachine.beginAnalysis();
 
             // plan query
+            final PlanVariableAllocator planVariableAllocator = new PlanVariableAllocator();
             LogicalPlanner logicalPlanner = new LogicalPlanner(
-                    false,
                     stateMachine.getSession(),
-                    planOptimizers,
                     idAllocator,
                     metadata,
+                    planVariableAllocator);
+
+            PlanNode planNode = getSession().getRuntimeStats().profileNanos(
```

Comment:
perhaps call this logicalPlanNode

Reply:
I feel planNode is sufficient, and its used at many other places. If we want to use logicalPlanNode then it would be better to change all other places as well. I will keep it planNode as of now, let me know if you feel strongly about it.